### DISCUSSION TITLE
how to use ffmpeg in vercel function?

### DISCUSSION THREAD

**Discussion by hackedbyxx** [vercel/vercel#9561] [Help] [answered]
i try to deploy my python server,but error No such file or directory: 'ffprobe'. it seems use aws serveless service，and i know aws can import ffmpeg enviroment,so how can i do it in vercel?

**Comment by wiiiimm** [comment, marked as answer]
While there are ways to get `ffmpeg` to install on Vercel's serverless containers like:

1. using the `@ffmpeg-installer/ffmpeg` npm package,
2. downloading `ffmpeg` as an executable binary (https://vercel.com/docs/concepts/deployments/build-image#installing-additional-packages) and optionally moving it to a cachable location

we generally do not recommend running this kind of server workload as serverless functions and installing `ffmpeg` will very likely push the serverless function over the [50MB limit](https://vercel.com/docs/concepts/limits/overview#serverless-function-size).

**Comment by ben-montra** [comment]
@williamli I have tried using the following script during both the build and install steps in my flask project to get a prebuilt binary installed

```sh
cd /usr/local/bin;
mkdir ffmpeg;
cd ffmpeg;
yum install wget tar xz;
wget https://johnvansickle.com/ffmpeg/releases/ffmpeg-release-amd64-static.tar.xz;
mkdir ffmpeg-release-amd64-static;
tar -xf ffmpeg-release-amd64-static.tar.xz --strip-components=1 -C ffmpeg-release-amd64-static;
ln -s /usr/local/bin/ffmpeg/ffmpeg-release-amd64-static/ffmpeg /usr/bin/ffmpeg;
ln -s /usr/local/bin/ffmpeg/ffmpeg-release-amd64-static/ffprobe /usr/bin/ffprobe;
```

This works well and you can see it working during the build/install by adding in `ffmpeg -version` into the relevant command. But when I invoke an endpoint in the api that has `print(os.system("ffmpeg -version"))`, I always get an error that ffmpeg can't be found.

I know this binary works in this particular image, how can i make sure it exists and is in the PATH of the serverless function.

Thanks
